test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriTemplateParserTests.cs

```diff
@@ -247,6 +247,18 @@ public void ParsePathTemplateSegmentWithTemplateParser()
             templateSegment.LiteralText.Should().Be("{some}");
         }
 
+        [Fact]
+        public void EntityReferenceCannotAppearAfterAPathTemplateSegment()

```

Comment:
>EntityReferenceCannotAppearAfterAPathTemplateSegment [](start = 20, length = 52)

why?

Reply:
I'm not too familiar with the PathTemplateSegment. This exception is thrown before we get to parsing $ref:

Microsoft.OData.UriParser.ODataUnrecognizedPathException: 'The request URI is not valid. The segment '{some}' must be the last segment in the URI because it is one of the following: $ref, $batch, $count, $value, $metadata, a named media resource, an action, a noncomposable function, an action import, a noncomposable function import, an operation with void return type, or an operation import with void return type.'